bar.go

```diff
@@ -89,7 +89,8 @@ func newBar(desk Desktop) fyne.CanvasObject {
 	if desk.WindowManager() != nil {
 		desk.WindowManager().AddStackListener(appBar)
 	}
-	for _, app := range appBar.desk.IconProvider().DefaultApps() {
+	for _, name := range Instance().Settings().DefaultApps() {
+		app := appBar.desk.IconProvider().FindAppFromName(name)
```

Comment:
For safety we should check that the app was found - it's possible that someone could edit the file (or sync it ;) ) and end up with a nil app here...